tests/warp/warp_test.go

```diff
@@ -426,66 +691,43 @@ var _ = ginkgo.Describe("[Warp]", ginkgo.Ordered, func() {
-		txBytes, err := signedTx.MarshalBinary()
-		gomega.Expect(err).Should(gomega.BeNil())
-		log.Info("Sending getVerifiedWarpBlockHash transaction", "txHash", signedTx.Hash(), "txBytes", common.Bytes2Hex(txBytes))
-		err = chainBWSClient.SendTransaction(ctx, signedTx)
-		gomega.Expect(err).Should(gomega.BeNil())
-
-		log.Info("Waiting for new block confirmation")
-		newHead := <-newHeads
-		blockHash := newHead.Hash()
-		log.Info("Fetching relevant warp logs and receipts from new block")
-		logs, err := chainBWSClient.FilterLogs(ctx, interfaces.FilterQuery{
-			BlockHash: &blockHash,
-			Addresses: []common.Address{warp.Module.Address},
-		})
-		gomega.Expect(err).Should(gomega.BeNil())
-		gomega.Expect(len(logs)).Should(gomega.Equal(0))
-		receipt, err := chainBWSClient.TransactionReceipt(ctx, signedTx.Hash())
-		gomega.Expect(err).Should(gomega.BeNil())
-		gomega.Expect(receipt.Status).Should(gomega.Equal(types.ReceiptStatusSuccessful))
-	})
-
-	ginkgo.It("Send Message from A to B from Hardhat", ginkgo.Label("Warp", "IWarpMessenger", "SendWarpMessage"), func() {
-		ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
-		defer cancel()
-		log.Info("Subscribing to new heads")
-		newHeads := make(chan *types.Header, 10)
-		sub, err := chainAWSClient.SubscribeNewHead(ctx, newHeads)
-		gomega.Expect(err).Should(gomega.BeNil())
-		defer sub.Unsubscribe()
-
-		rpcURI := toRPCURI(chainAURIs[0], blockchainIDA.String())
-		senderAddress := common.HexToAddress("0x8db97C7cEcE249c2b98bDC0226Cc4C2A57BF52FC")
-		addressedPayload, err := payload.NewAddressedCall(
-			senderAddress.Bytes(),
-			testPayload,
-		)
-		gomega.Expect(err).Should(gomega.BeNil())
-		expectedUnsignedMessage, err := avalancheWarp.NewUnsignedMessage(
-			1337,
-			blockchainIDA,
-			addressedPayload.Bytes(),
+			signedWarpMessageBytes,
 		)
-		gomega.Expect(err).Should(gomega.BeNil())
-
-		os.Setenv("SENDER_ADDRESS", senderAddress.Hex())
-		os.Setenv("SOURCE_CHAIN_ID", "0x"+blockchainIDA.Hex())
-		os.Setenv("PAYLOAD", "0x"+common.Bytes2Hex(testPayload))
-		os.Setenv("EXPECTED_UNSIGNED_MESSAGE", "0x"+hex.EncodeToString(expectedUnsignedMessage.Bytes()))
-
-		cmdPath := "./contracts"
-		// test path is relative to the cmd path
-		testPath := "./test/warp.ts"
-		utils.RunHardhatTestsCustomURI(ctx, rpcURI, cmdPath, testPath)
-	})
-})
+		return types.SignTx(tx, w.receivingSubnetSigner, key)
+	}, w.receivingSubnetClients[0], privateKeys, txsPerWorker, true)
+	require.NoError(err)
+	log.Info("Executing warp delivery...")
+	warpDeliverLoader := load.New(chainBWorkers, warpDeliverSequences, batchSize, loadMetrics)
+	require.NoError(warpDeliverLoader.Execute(ctx))
+	require.NoError(warpSendLoader.ConfirmReachedTip(ctx))
+	log.Info("Completed warp delivery successfully.")
+}
 
 func toRPCURI(uri string, blockchainID string) string {
 	return fmt.Sprintf("%s/ext/bc/%s/rpc", uri, blockchainID)
 }
+
+var _ = ginkgo.DescribeTable("[Warp]", func(gen func() *warpTest) {
```

Comment:
Could we move this up next to where `warpTableEntries` is defined?

Reply:
Moving to the top